modules/light-clients/08-wasm/types/client_state_test.go

```diff
@@ -276,66 +276,83 @@ func (suite *TypesTestSuite) TestInitializeGrandpa() {
 	}
 }
 
-// func (suite *TypesTestSuite) TestInitializeTendermint() {
-// 	var consensusState exported.ConsensusState
-// 	testCases := []struct {
-// 		name     string
-// 		malleate func()
-// 		expPass  bool
-// 	}{
-// 		{
-// 			name: "valid consensus",
-// 			malleate: func() {
-// 				tmConsensusState := tmtypes.NewConsensusState(time.Now(), commitmenttypes.NewMerkleRoot([]byte{0}), []byte(codeHash))
-// 				tmConsensusStateData, err := suite.chainA.Codec.MarshalInterface(tmConsensusState)
-// 				suite.Require().NoError(err)
+func (suite *TypesTestSuite) TestInitialize() {
+	panicMsg := errors.New("panic in InstantiateFn")
```

Comment:
I'm unsure if we should bother testing for a panic... what do you think? We should never have to handle a panic from the vm inside of our code, right? Aren't all panics caught [here](https://github.com/CosmWasm/wasmvm/blob/main/internal/api/callbacks.go#L56) within the actual vm and propagated as errors?

Would probably simplify the test structure if we didn't have to include this, but either way test looks good!

Reply:
Since wasm engine is an interface now, I think it should be in our scope to include basic tests that might break certain assumptions about wasmvm. Also, I'm not sure if they panic later again somewhere in their code

Reply:
I'd agree with Damian here. From what I can understand, the idea is to test that panics are propagated across the boundary (which might not be FFI like wasmvm)? I'm trying to understand the benefit of testing for this, especially given that wasmvm appears to translate em all into err's i.e if we plugged wasmvm here, it would fail.

W/e the case, happy to keep on this path if there's consensus!

Reply:
cc. @chatton @charleenfei @colin-axner any thoughts/preference? I didn't implement panic tests in #4923 yet but if there's consensus to do it I will push an update. 

I'd see more of an argument to testing for panics if we were actually recovering from them in client state methods, but we aren't, if a panic occurs across the api boundary here its just going to propagated to baseapp anyways. 
I'd be happy to skip the panic tests as it simplifies the structure and is easier to read and maintain in the long run

Reply:
Yeah, I guess the only benefit I see to implementing panic tests is making sure that there are no panic recoveries anywhere in the stack between callbackFn and 02-client cc. @damiannolan

Reply:
sorry late to this thread, but i told @damiannolan my thoughts in person (and [in my original discussion on this topic](https://github.com/cosmos/ibc-go/pull/4901#discussion_r1365545002) with @srdtrk)